spynnaker/pyNN/models/neuron/synapse_dynamics/synapse_dynamics_weight_changable.py

```diff
@@ -0,0 +1,390 @@
+    AbstractConnector)
+from spynnaker.pyNN.types import Weight_Types
+from spynnaker.pyNN.types import Weight_Delay_In_Types as _In_Types
+from spynnaker.pyNN.utilities.utility_calls import get_n_bits
+from spynnaker.pyNN.models.neuron.synapse_dynamics.types import (
+    NUMPY_CONNECTORS_DTYPE)
+from .abstract_plastic_synapse_dynamics import AbstractPlasticSynapseDynamics
+from .abstract_generate_on_machine import (
+    AbstractGenerateOnMachine, MatrixGeneratorID)
+
+if TYPE_CHECKING:
+    from spynnaker.pyNN.models.neural_projections import (
+        ProjectionApplicationEdge, SynapseInformation)
+    from spynnaker.pyNN.models.neuron.synapse_dynamics.types import (
+        ConnectionsArray)
+    from spynnaker.pyNN.models.neuron.synapse_io import MaxRowInfo
+    from .abstract_synapse_dynamics import AbstractSynapseDynamics
+
+# How large are the time-stamps stored with each event
+TIME_STAMP_BYTES = BYTES_PER_WORD
+
+# The targets of neuromodulation
+NEUROMODULATION_TARGETS = {
+    "reward": 0,
+    "punishment": 1
+}
+
+
+class SynapseDynamicsWeightChangable(
+        AbstractPlasticSynapseDynamics, AbstractGenerateOnMachine):
+    """
+    The dynamics of a synapse that can be changed simply by the sending of an
+    external signal.
+    """
+
+    __slots__ = (
+
+        # The maximum weight
+        "__weight_max",
+
+        # The minimum weight
+        "__weight_min",
+
+        # The map of synapse information to index
+        "__synapse_info_to_index",
+
+        # The next index to use for the next projection
+        "__next_index")
+
+    def __init__(
+            self,
+            weight_min: float, weight_max: float,
+            weight: _In_Types = StaticSynapse.default_parameters['weight'],
+            delay: _In_Types = None):
+        """
+        :param float weight:
+        :param delay: Use ``None`` to get the simulator default minimum delay.
+        :type delay: float or None
+        """
+        super().__init__(delay=delay, weight=weight)
+        self.__weight_max = weight_max
+        self.__weight_min = weight_min
+        self.__synapse_info_to_index: Dict[SynapseInformation, int] = dict()
+        self.__next_index = 0
+
+    @property
+    def weight_max(self) -> float:
+        """ Get the maximum weight allowed to change to
+        """
+        return self.__weight_max
+
+    @property
+    def weight_min(self) -> float:
+        """ Get the minimum weight allowed to change to
+        """
+        return self.__weight_min
+
+    def get_synapse_info_index(self, synapse_info: SynapseInformation) -> int:
+        """ Get the row offset for the given synapse information.  Each synapse
+            information has a unique row offset which then allows for multiple
+            connections to be identified and kept separate.
+        """
+        if synapse_info not in self.__synapse_info_to_index:
+            self.__synapse_info_to_index[synapse_info] = self.__next_index
```

Comment:
self.__synapse_info_to_index[synapse_info] = len(self.__synapse_info_to_index) is easier and safer

Reply:
But that is incorrect - the index is based on the number of atoms previously added, not the number of connections